patches/server/0952-fixed-null-pointer-exception-when-invoking-Allay-sto.patch

```diff
@@ -0,0 +1,25 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: NonSwag <mrminecraft00@gmail.com>
+Date: Sun, 4 Dec 2022 13:58:15 +0100
+Subject: [PATCH] fixed null pointer exception when invoking Allay#stopDancing
+ without an active jukebox
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftAllay.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftAllay.java
+index db3b0a9d3696f31d473342312d437e1eae97404a..29b25ee3081060b53358683a193196c01c80afb4 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftAllay.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftAllay.java
+@@ -82,8 +82,11 @@ public class CraftAllay extends CraftCreature implements org.bukkit.entity.Allay
+     @Override
+     public void stopDancing() {
+         this.getHandle().forceDancing = false;
+-        this.getHandle().jukeboxPos = null;
```

Comment:
why did you remove this line?

Reply:
the setJukeboxPlaying already set null the jukebox.. but see again the changes maybe its better only make the jukebox null and in the next tick the aiStep when check the jukebox stop the dance... make a thing like..
```java
@Override
public void stopDancing() {
  getHandle().forceDancing = false; // cancel the CB field for force dance
  getHandle().jukeboxPos = null; // remove any jukebox
  getHandle().setDancing(false); // save the nbt for dance in false
 }
```
this like the same than the setJukebox used in the original implementation.